src/hotspot/share/jfr/leakprofiler/checkpoint/objectSampleWriter.cpp

```diff
@@ -612,11 +613,6 @@ ObjectSampleWriter::ObjectSampleWriter(JfrCheckpointWriter& writer, EdgeStore* s
   assert(store != NULL, "invariant");
   assert(!store->is_empty(), "invariant");
   register_serializers();
-  sample_infos = NULL;
-  ref_infos = NULL;
-  array_infos = NULL;
-  field_infos = NULL;
-  root_infos = NULL;
```

Comment:
If the assumption is that only one Writer uses these global variables at a time, I'd be happier with some asserts here.